### PR TITLE
fixing auth reconcile command for 3.10

### PR DESCRIPTION
**Pull request by kalexand-rh**
https://trello.com/c/w639uojU/901-document-authz-rbac-migration-followups-for-38

@mrogers950, does this sound right to you?

### CODE REVIEW

File: upgrading/manual_upgrades.adoc [hunk 1]
```diff
@@ -507,80 +507,41 @@ updated.
...
+====
+
+. Use the policy file to automatically reconcile the roles and role bindings:
 +
 ----
-# oc adm policy reconcile-cluster-role-bindings \
-    system:build-strategy-jenkinspipeline \
-    --confirm \
-    -o name
+$ oc process --local -f policy.json | oc auth reconcile -f -
```
Comment (mrogers950) [inline upgrading/manual_upgrades.adoc]: The policy file changed to no longer be a "template" type so this full command is now `oc auth reconcile -f policy.json`

File: upgrading/manual_upgrades.adoc [hunk 2]
```diff
@@ -507,80 +507,41 @@ updated.
...
-made, or you can automatically apply the new policy using the following process:
 
-. Reconcile the cluster roles:
-+
-----
-# oc adm policy reconcile-cluster-roles \
-    --additive-only=true \
-    --confirm
-----
+To check if all defaults are set as recommended for your environment, run the
```
Comment (mrogers950) [inline upgrading/manual_upgrades.adoc]: Until the reconcile command has a proper --dry-run setting in a later release, what follows is not a "check" but will actually reset the cluster roles and rolebindings to their default values. Custom changes to the cluster roles would need to be applied to the policy file before running reconcile (or changed manually afterwards).

**Comment by mrogers950** [comment]
cc @enj

File: upgrading/manual_upgrades.adoc [hunk 3]
```diff
@@ -507,80 +507,41 @@ updated.
...
-    --additive-only=true \
-    --confirm
+$ oc adm create-bootstrap-policy-file --filename=policy.json
 ----
 +
-Also run:
+[NOTE]
+====
+The contents of the template file vary based on the OpenShift version and any local
+customizations you made. Review the proposed policy carefully.
```
[the remark this answers is not in the document]
Comment (kalexand-rh) [reply upgrading/manual_upgrades.adoc]: Is there an easy way to tell if you've made local customizations to the policies that you need to retrieve and add to the file?

**Comment by kalexand-rh** [comment]
@mrogers950, @enj, PTAL

File: upgrading/manual_upgrades.adoc [hunk 4]
```diff
@@ -507,80 +507,38 @@ updated.
...
-+
-Also run:
+. Use the policy file to automatically reconcile the roles and role bindings:
 +
 ----
-# oc adm policy reconcile-cluster-role-bindings \
-    system:build-strategy-jenkinspipeline \
-    --confirm \
-    -o name
+$ oc process --local -f policy.json | oc auth reconcile -f policy.json
```
Comment (mrogers950) [inline upgrading/manual_upgrades.adoc]: The whole command here should be just the part after the pipe;  `oc auth reconcile -f policy.json`

**Comment by kalexand-rh** [comment]
@xltian, will you PTAL?
@openshift/team-documentation, will you please review?

**Comment by xltian** [comment]
@stuartchuan will review from QE side.

File: upgrading/manual_upgrades.adoc [hunk 5]
```diff
@@ -507,80 +507,38 @@ updated.
...
 ====
-If you have customized default cluster roles and want to ensure a role reconciliation
-does not modify those customized roles, annotate them with `openshift.io/reconcile-protect`
-set to `true` when using the old Openshift policy format. When using the new RBAC
-roles, use `rbac.authorization.kubernetes.io/autoupdate` set to `false` instead.
-In doing so, you are responsible for manually updating those roles with any new
-or required permissions during upgrades.
+If you customized default cluster roles and want to ensure a role reconciliation
+does not modify those customized roles, set the
+`rbac.authorization.kubernetes.io/autoupdate` parameter to `false`.
```
Comment (barleyer) [inline upgrading/manual_upgrades.adoc]: @enj ,do we use `openshift.io/reconcile-protect` now? i see `convert_rbac_Annotations_To_authorization_Annotations`.

Comment (barleyer) [reply upgrading/manual_upgrades.adoc]: and i cann't add `rbac.authorization.kubernetes.io/autoupdate` to clusterrole's `annotations`.i use *openshift v3.10.0-0.66.0*

Comment (barleyer) [reply upgrading/manual_upgrades.adoc]: maybe,that's not a problem.

File: architecture/additional_concepts/authorization.adoc [hunk 6]
```diff
@@ -430,14 +430,12 @@ ifdef::openshift-enterprise,openshift-origin[]
...
-server is started. Additionally, see
-xref:../../upgrading/manual_upgrades.adoc#updating-policy-definitions[Updating
-Policy Definitions] for instructions on getting other recommendations
-using:
+server is started. 
 
-----
-$ oc adm policy reconcile-cluster-roles
-----
+If you customize the default roles, you must
```
Comment (enj) [inline architecture/additional_concepts/authorization.adoc]: If you customize and reconcile protect

File: upgrading/manual_upgrades.adoc [hunk 7]
```diff
@@ -507,80 +507,38 @@ updated.
 [[updating-policy-definitions]]
 == Updating Policy Definitions
 
-After a cluster upgrade, the default roles
+During a cluster upgrade, the default roles
```
Comment (enj) [inline upgrading/manual_upgrades.adoc]: During a cluster upgrade and on every restart of any master

File: upgrading/manual_upgrades.adoc (hunk 5 above)
Comment (enj) [inline upgrading/manual_upgrades.adoc]: This needs a clear `oc` example on how to annotate RBAC (i.e. `clusterrole.rbac` vs `clusterrole`).

File: upgrading/manual_upgrades.adoc [hunk 8]
```diff
@@ -507,80 +507,38 @@ updated.
...
-+
-Also run:
+. Use the policy file to automatically reconcile the roles and role bindings:
 +
 ----
-# oc adm policy reconcile-cluster-role-bindings \
-    system:build-strategy-jenkinspipeline \
-    --confirm \
-    -o name
+$ oc auth reconcile -f policy.json
```
Comment (enj) [inline upgrading/manual_upgrades.adoc]: This will not work for reconcile protected roles.

File: upgrading/manual_upgrades.adoc (hunk 5 above)
Comment (enj) [reply upgrading/manual_upgrades.adoc]: Use `oc get clusterrole.rbac -o yaml` to see what the "real" object has.

**Review by enj — CHANGES_REQUESTED** [review]

File: upgrading/manual_upgrades.adoc (hunk 5 above)
Comment (barleyer) [reply upgrading/manual_upgrades.adoc]: @enj ,thanks.

File: upgrading/manual_upgrades.adoc (hunk 7 above)
Comment (barleyer) [reply upgrading/manual_upgrades.adoc]: @enj, can we edit the `policy.json` which restart master use?

File: upgrading/manual_upgrades.adoc (hunk 5 above)
Comment (mrogers950) [reply upgrading/manual_upgrades.adoc]: ex: `oc annotate clusterrole.rbac system:webhook --overwrite rbac.authorization.kubernetes.io/autoupdate=false`

File: upgrading/manual_upgrades.adoc [hunk 9]
```diff
@@ -507,80 +507,38 @@ updated.
...
-    --exclude-groups=system:authenticated \
-    --exclude-groups=system:authenticated:oauth \
-    --exclude-groups=system:unauthenticated \
-    --exclude-users=system:anonymous \
-    --additive-only=true \
-    --confirm
-----
-+
-Also run:
+. Use the policy file to automatically reconcile the roles and role bindings:
```
Comment (mrogers950) [inline upgrading/manual_upgrades.adoc]: "Use the policy file to automatically reconcile roles and role bindings that are not reconcile protected"

File: upgrading/manual_upgrades.adoc (hunk 7 above)
Comment (kalexand-rh) [reply upgrading/manual_upgrades.adoc]: @enj? ^

**Comment by kalexand-rh** [comment]
@mrogers950, will you PTAL?

**Comment by kalexand-rh** [comment]
@openshift/team-documentation, PTAL

File: architecture/additional_concepts/authorization.adoc [hunk 10]
```diff
@@ -430,14 +430,12 @@ ifdef::openshift-enterprise,openshift-origin[]
...
-xref:../../upgrading/manual_upgrades.adoc#updating-policy-definitions[Updating
-Policy Definitions] for instructions on getting other recommendations
-using:
+server is started. 
 
-----
-$ oc adm policy reconcile-cluster-roles
-----
+If you customize and reconcile protect the default roles, you must
+xref:../../upgrading/manual_upgrades.adoc#updating-policy-definitions[manually
```
Comment (mburke5678) [inline architecture/additional_concepts/authorization.adoc]: @kalexand-rh Did you mean _reconcile protect_? Not sure what that means.

File: upgrading/manual_upgrades.adoc [hunk 11]
```diff
@@ -507,80 +507,42 @@ updated.
...
-After a cluster upgrade, the default roles
+During a cluster upgrade, and on every restart of any master, the 
 xref:../architecture/additional_concepts/authorization.adoc#roles[default
-cluster roles] are automatically updated. To check if all defaults are set as
-recommended for your environment, run:
+cluster roles] are automatically updated. 
 
+. If you customized default cluster roles and want to ensure a role reconciliation
+does not modify them, protect each role from reconciliation:
++
```
Comment (mburke5678) [inline upgrading/manual_upgrades.adoc]: @kalexand-rh Do we need to define _reconciliation_ here?  Not sure we define it anywhere.

File: upgrading/manual_upgrades.adoc [hunk 12]
```diff
@@ -507,80 +507,42 @@ updated.
...
 ----
-
+$ oc adm create-bootstrap-policy-file --filename=policy.json
+----
++
 [NOTE]
 ====
-Your output will vary based on the OpenShift version and any local
-customizations you have made. Review the proposed policy carefully.
+The contents of the file vary based on the OpenShift version, but the file
```
Comment (mburke5678) [inline upgrading/manual_upgrades.adoc]: s/OpenShift/{product-title} ?

**Comment by mburke5678** [comment]
@kalexand-rh Couple o' comments.

**Comment by kalexand-rh** [comment]
Thanks @mburke5678!

@enj, @mrogers950, will one of you PTAL at the most recent commit?

File: upgrading/manual_upgrades.adoc [hunk 13]
```diff
@@ -509,7 +509,8 @@ updated.
 
 During a cluster upgrade, and on every restart of any master, the 
 xref:../architecture/additional_concepts/authorization.adoc#roles[default
-cluster roles] are automatically updated. 
+cluster roles] are automatically reconciled, or updated to contain all the
```
Comment (mrogers950) [inline upgrading/manual_upgrades.adoc]: This seems to imply that it does one or the other - perhaps we can say
"are automatically reconciled to restore any missing permissions.

**Comment by kalexand-rh** [comment]
Per discussion around https://bugzilla.redhat.com/show_bug.cgi?id=1608121, @mrogers950 says that this change should apply to 3.7 and 3.9, too.

**Pull request merged**